sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/externalreport/AbstractReportSensor.java

```diff
@@ -94,11 +100,42 @@ RuleType ruleType(String key) {
   public void describe(SensorDescriptor sensorDescriptor) {
     sensorDescriptor
       .onlyOnLanguage(TypeScriptLanguage.KEY)
+      .onlyWhenConfiguration(conf -> conf.hasKey(reportsPropertyName()))
       .name("Import of " + linterName() + " issues");
   }
 
+  static void createExternalRuleRepository(Context context, String repositoryKey, String repositoryName, Set<String> bugRules) {
+    NewRepository externalRepo = context.createExternalRepository(repositoryKey, TypeScriptLanguage.KEY).setName(repositoryName);
+    String pathToRulesMeta = "org/sonar/l10n/typescript/rules/" + repositoryKey + "/rules.json";
+    String description = "See the description of %s rule <code>%s</code> at <a href=\"%s\">%s website</a>.";
+
+    try (InputStreamReader inputStreamReader = new InputStreamReader(AbstractReportSensor.class.getClassLoader().getResourceAsStream(pathToRulesMeta), StandardCharsets.UTF_8)) {
+      ExternalRule[] rules = gson.fromJson(inputStreamReader, ExternalRule[].class);
+      for (ExternalRule rule : rules) {
+        NewRule newRule = externalRepo.createRule(rule.name).setName(rule.description);
```

Comment:
It's confusing. You map:
rule.name into newRule.key
rule.description into newRule.name
And "description" is not the description but the description format.